src/app/components/button/button.component.html

```diff
@@ -0,0 +1 @@
+<button mat-raised-button [disabled]="disabled"> {{ buttonText }}</button>
```

Comment:
why do we need this component?

Reply:
If it's just for the styling part, you should create just a CSS class that will be applied to that button

Reply:
TL;DR: I just deleted the button 

The initial idea was that a separate component can be used in another components from other pages ( for example the login page, but provided with the corresponding label and event ). Now when I think about it, it is kind of an overkill because it was not as generic as it should've been for a general purpose button.

Also, it had an additional flaw: the event was bound in the parent component, not the button itself, and that lead to a bug (event function called by itself when the enable/disable attribute was changed). I think a solution for that is to make the button (child component) emitting the event on click and catch it in the parent, but that also seems a little bit too much.

So, I just delete the component and added a button as an element in the register card and now it is fine :)